### PR TITLE
D04 completed

### PR DESCRIPTION
**Pull request by taherkheli**

### CODE REVIEW

File: src/D04/Pair.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿namespace aoc.D04
+{
+  public struct Pair
```
Comment (Jaxwood) [inline src/D04/Pair.cs]: @taherkheli you could use  the new record type :D it has some nice properties like equality comparison

File: src/D04/PairAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
+﻿namespace aoc.D04
+{
+  public static class PairAnalyzer
+  {
+    public static int GetFullyCotainedPairsCount(List<Pair> pairs)
```
Comment (Jaxwood) [inline src/D04/PairAnalyzer.cs]:
```suggestion
    public static int GetFullyCotainedPairsCount(IEnumerable<Pair> pairs)
```

File: src/D04/PairAnalyzer.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    {
+      int count = 0;
+
+      foreach (Pair pair in pairs)
```
Comment (Jaxwood) [inline src/D04/PairAnalyzer.cs]:
```suggestion
      foreach (var pair in pairs)
```

File: src/D04/Parser.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  public static class Parser
+  {
+    public static List<Pair> Parse(string path)
+    {
+      var pairs = new List<Pair>();
+      var lines = File.ReadAllLines(path);
+
+      foreach (var line in lines)
+      {
+        var elf1 = new List<int>();
```
Comment (Jaxwood) [inline src/D04/Parser.cs:12]: here you use `var` other places not - wonder what is your criteria 🤔

File: test/D04/TC01.cs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    [Fact]
+    public void InitialI()
+    {
+      int expected = 2;
+      var pairs = Parser.Parse(@"./D04/initial.txt");
+      var actual = PairAnalyzer.GetFullyCotainedPairsCount(pairs);
+      Assert.Equal(expected, actual);
+    }
+
+    [Fact]
```
Comment (Jaxwood) [inline test/D04/TC01.cs:17]: you could use `[Theory]` and `[InlineData]`

File: test/D04/TC01.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    public void PartII()
+    {
+      int expected = 867;
+      var pairs = Parser.Parse(@"./D04/input.txt");
+      var actual = PairAnalyzer.GetOverlappingPairsCount(pairs);
+      Assert.Equal(expected, actual);
+    }
+  }
+}
```
Comment (Jaxwood) [inline test/D04/TC01.cs]: missing newline

File: src/D04/Pair.cs [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+  {
+    public List<int> Elf1;
```
Comment (Jaxwood) [inline src/D04/Pair.cs]:
```suggestion
    public IEnumerable<int> Elf1;
```

File: src/D04/Pair.cs [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+    public List<int> Elf2;
```
Comment (Jaxwood) [inline src/D04/Pair.cs]:
```suggestion
    public IEnumerable<int> Elf2;
```

**Review by Jaxwood — APPROVED** [review]

File: test/D04/TC01.cs (hunk 5 above)
Comment (taherkheli) [reply test/D04/TC01.cs:17]: does not help much, inputs are different called methods are different for partI n II

**Comment by taherkheli** [comment]
@Jaxwood happy now?

**Pull request merged**